airbyte-cdk/python/airbyte_cdk/sources/file_based/config/unstructured_format.py

```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from pydantic import BaseModel, Field
+
+
+class UnstructuredFormat(BaseModel):
+    class Config:
+        title = "Markdown/PDF/Docx Format (Experimental)"
+        schema_extra = {"description": "Extract text from document formats and emit as one record per file."}
+
+    filetype: str = Field(
+        "unstructured",
+        const=True,
+    )
```

Comment:
@flash1293 - What do you think of adding these as n different formats?

It feels like in the long-run, it may be clearer to have just a selector for MS Word, PDF, Markdown, Plain Text, etc.

This more closely mirrors how Excel, CSV, JSONL, etc., and it keeps the file format choice decoupled from our implementation code path.

That said, I also understand there may also be some config shared across these types, but we can still model these config options in a DRY way via pydantic. 

This also would imply distinct parser classes subclassing from an Unstructured parser base class

Reply:
Circling back on this - the `title` has since been updated to be nicely generic, and we discussed in DM. 

We're agreed a single parser makes sense in this case, and will be ideal to pick up a combination of text document file types.